src/main/java/pl/betse/beontime/timesheet/service/TimeEntryService.java

```diff
@@ -0,0 +1,233 @@
+                    log.error("Time entry for user " + timeEntryEntity.getUserGuid() + " and project " + timeEntryEntity.getProjectGuid() + " with date " + timeEntryEntity.getEntryDate() + " currently NOT exist in database!");
+                    throw new TimeEntryNotFound();
+                }
+            }
+        });
+    }
+
+    public void checkIfDateIsInCorrectWeekOfYear(List<TimeEntryBo> timeEntryBoList, String week) {
+        int weekNumber = Integer.parseInt(week.substring(6, 8));
+        timeEntryBoList.forEach(entry -> {
+            if (entry.getEntryDate().get(IsoFields.WEEK_OF_WEEK_BASED_YEAR) != weekNumber) {
+                log.error("Date is not in that week of the year.");
+                throw new BadWeekAndDateException();
+            }
+        });
+    }
+
+    public void checkIfDateHasCorrectMonth(List<TimeEntryBo> timeEntryBoList, LocalDate requestedDate) {
+        int monthNumber = requestedDate.getMonthValue();
+        timeEntryBoList.forEach(entry -> {
+            if (entry.getEntryDate().getMonthValue() != monthNumber) {
+                log.error("Date is not in that month of the year.");
+                throw new BadMonthAndDateException();
+            }
+        });
+    }
+
+    public void verifyStatusesBeforeCreatingNewEntry(List<TimeEntryBo> timeEntryBoList) {
+        timeEntryBoList.forEach(entry -> {
+            if (!entry.getStatus().equalsIgnoreCase(SAVED_STATUS)) {
+                String message = ": Only entries with '" + SAVED_STATUS + "' status can be created as new.";
+                log.error(message);
+                throw new WrongStatusException(message);
+            }
+        });
+    }
+
+    public void verifyStatusesBeforeDeleting(List<TimeEntryBo> timeEntryBoList) {
+        timeEntryBoList.forEach(entry -> {
+            if (!entry.getStatus().equalsIgnoreCase(SAVED_STATUS)) {
+                String message = ": Only entries with '" + SAVED_STATUS + "' status can be deleted.";
+                log.error(message);
+                throw new WrongStatusException(message);
+            }
+        });
+    }
+
+    public void verifyStatusesBeforeAddingComment(List<TimeEntryBo> timeEntryBoList) {
+        timeEntryBoList.forEach(entry -> {
+            if (!entry.getStatus().equalsIgnoreCase(REJECTED_STATUS) && !entry.getComment().isEmpty()) {
+                String message = ": Only entries with '" + REJECTED_STATUS + "' status can be commented.";
+                log.error(message);
+                throw new WrongStatusException(message);
+            }
+        });
+    }
+
+    private void resolveProjectGuidFromBoList(List<TimeEntryBo> timeEntryBoList, List<String> projectsGuid) {
+        timeEntryBoList.forEach(entry -> {
+            if (!projectsGuid.contains(entry.getProjectGuid())) {
+                projectsGuid.add(entry.getProjectGuid());
```

Comment:
please don't change object which you pass a parameter - it is very unclear to debug and read